DEPLOYING.md

```diff
@@ -0,0 +1,126 @@
+This includes the PDC service ("web") and other services upon which it depends.
+
+How do they get deployed? A [bash script running under cron][3] gets the new
+compose file, pulls new components (images) using the new compose file, stops
+the old services (containers) using the old compose file, and starts the new
+services (containers) using the new compose file.
+
+How does the deploy script know how what to deploy? It [looks inside a file][4].
+
+This setup supports both automated and manual deployments that have exactly the
+same steps with exactly the same code.
+
+[1]: https://app.netlify.com/sites/philanthropy-data-commons-viewer/settings
+[2]: https://github.com/PhilanthropyDataCommons/deploy/blob/main/compose.yml
+[3]: https://github.com/PhilanthropyDataCommons/deploy/blob/main/deploy.sh
+[4]: https://github.com/PhilanthropyDataCommons/deploy/blob/e0d24b8a19ee563938f8211bca0485a97ff1a6e9/deploy.sh#L90
+
+## To deploy a newly built version of the PDC service to production
+
+This is the streamlined case. It assumes that both the test and production
+machines were already set up with users, directories, docker, certbot, scripts,
+environment variables, additional keycloak jar files, and so forth.
+
+### Optionally review this background information
+
+When someone merges code to the main branch in the [service repository][5],
+[GitHub Actions][6] runs tests, builds a new docker image with the PDC service
+and its dependencies, and pushes the new image to [service packages][7]. The
+[GitHub Actions][6] also tag the new service image with a version, such as
+`20230616-00a3cc8`. The tag is on the binary, not the source code, but the
+version text is based on the git commit, so that we can always trace the image
+back back to the source code whence it came.
+
+The [github action that builds and pushes an image][8] also triggers an [action
+in the deploy repository][9] which in turn updates the [compose.yml][2] with the
+newly created image version of the service, tags the [compose.yml][2] with a
+version, and [automatically deploys][10] PDC software to the test environment
+using the newly updated and tagged [compose.yml][2].
+
+The rest of this guide assumes you saw that tests passed and you want to deploy
+the recently built and tested version of the PDC service to the production
+environment.
+
+[5]: https://github.com/PhilanthropyDataCommons/service
+[6]: https://github.com/PhilanthropyDataCommons/service/tree/main/.github/workflows
+[7]: https://github.com/PhilanthropyDataCommons/service/pkgs/container/service
+[8]: https://github.com/PhilanthropyDataCommons/service/blob/cd2e2a4c39d87479f7b145a9163e155ede7676f2/.github/workflows/build.yml#L46
+[9]: https://github.com/PhilanthropyDataCommons/deploy/blob/main/.github/workflows/update-service-image.yml
+[10]: https://github.com/PhilanthropyDataCommons/deploy/blob/e0d24b8a19ee563938f8211bca0485a97ff1a6e9/.github/workflows/send-tag-to-machine.yml#L22
```

Comment:
Yeah it is a pain to update all the link numbers. I'll replace them with brief descriptive text.

Reply:
One of the advantages of the numbers was to prevent so many awkward line breaks and spilling-over-the-80-char mark. I can't seem to avoid that with the other style but I agree it's going to be more maintainable to keep the links in-band right at the point of reference.

Reply:
I have found [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) to be an effective practice for formatting Markdown!

Reply:
I am sure it helps maintainability but the result is really ugly when trying to read in plain text.

Reply:
I would like to gently suggest that this is an instance where it is worthwhile to intentionally develop your taste. Maintainable code - and prose! - should be aesthetically pleasing, even if it takes some work to get there. :slightly_smiling_face: